### PR TITLE
build: make tests run faster

### PR DESCRIPTION
**Pull request by sofisl**
Partially addresses #4140 

- [x] Previous builds are cancelled when new builds are started on the same branch
- [x] MinifyProtos is moved to postpack for publishing, removed from running before test execution; sample & system tests do not compile before running, since the `prepare` script will do that as well
- [x] Switch to using github actions for unit tests and `pnpm` instead of `npm`
- [x] Check returns immediately if it fails

### CODE REVIEW

File: make-change-to-package.cjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const fs = require('fs');
+const path = require('path');
+const cp = require('child_process');
+const dirsToSkip = ['gapic-node-templating', 'typeless-sample-bot'];
```
Comment (bcoe) [inline make-change-to-package.cjs:4]: what purpose does this script serve?

File: packages/google-maps-routing/package.json [hunk 2]
```diff
@@ -57,6 +58,7 @@
     "jsdoc-fresh": "^2.0.0",
     "jsdoc-region-tag": "^2.0.0",
     "linkinator": "^4.0.0",
+    "long": "^5.2.3",
```
Comment (bcoe) [inline packages/google-maps-routing/package.json:61]: I don't think we should need to add this `long` dependency, as it's pulled in by google-gax. I'm curious why this was needed.

File: packages/google-cloud-beyondcorp-clientconnectorservices/samples/test/quickstart.js
Comment (sofisl) [inline packages/google-cloud-beyondcorp-clientconnectorservices/samples/test/quickstart.js:1]: commenting these tests out per b/278007126

File: make-change-to-package.cjs (hunk 1 above)
Comment (sofisl) [reply make-change-to-package.cjs:4]: I will delete it! Shouldn't be in here.

File: packages/google-maps-routing/package.json (hunk 2 above)
Comment (sofisl) [reply packages/google-maps-routing/package.json:61]: The way pnpm installs dependencies doesn't grab this dependency. It's really only for [this line](https://github.com/googleapis/google-cloud-node/blob/8db807ae80a695bb1ad4fa290bc9a50ce29a8365/packages/google-api-servicemanagement/protos/protos.d.ts#L16). I spoke with @alexander-fenster about this, and he recommended this option, over skipping type checking in the tsconfig.

**Review by bcoe — APPROVED** [review]
ship it!

**Pull request merged**